crates/router/src/core/payments/flows/authorize_flow.rs

```diff
@@ -339,10 +339,16 @@ impl Feature<api::Authorize, types::PaymentsAuthorizeData> for types::PaymentsAu
                     .connector
                     .validate_capture_method(
                         self.request.capture_method,
+                        self.payment_method,
                         self.request.payment_method_type,
                     )
                     .to_payment_failed_response()?;
 
+                connector
+                    .connector
+                    .validate_payment_method(self.request.payment_method_type, self.payment_method)
```

Comment:
you can remove this since the validation of payment method is already happening in the `validate_capture_method` function.